_posts/2023-12-20-in-progress-wildfly-feature-on-openshift.adoc

```diff
@@ -0,0 +1,312 @@
+
+=== Clone the WildFly and Elytron Projects
+First we need to clone the `WildFly` and the `WildFly-Elytron` projects with the branches that contain the OIDC scope feature: 
+```
+git clone https://github.com/PrarthonaPaul/wildfly.git
+git checkout WFLY-16532
+```
+Next, go to the directory where the `WildFly-Elytron` project should be cloned and clone it:
+```
+git clone https://github.com/PrarthonaPaul/wildfly-elytron.git
+git checkout WFLY-16532
+```
+Notice that we are not using `WildFly/WildFly` and the `wildfly-security/wildfly-elytron` repositories. Instead, we are
+using a forked versions of the projects. This feature does not add any changes to the `WildFly-Core` project. However,
+to ensure that the branch for `WildFly-Core` is compatible with the branches in the `WildFly` and the `WildFly-Elytron`
+projects, we have created a branch called `WFLY-16532`, which we will use for this guide.
+```
+git clone https://github.com/PrarthonaPaul/wildfly-core.git
+git checkout WFLY-16532
+```
+
+=== Build the WildFly Project
+Next, we need to build these projects to use the local version of the code. To do this, we will specify the SNAPSHOT versions of `Elytron` and `WildFly-core`: 
+```
+cd wildfly-elytron
+mvn clean install -DskipTests
+
+cd ../wildfly-core
+mvn clean install -DskipTests -Dversion.org.wildfly.security.elytron=<insert_SNAPSHOT_VERSION_OF_ELYTRON> 
+
+cd ../wildfly
+mvn clean install -DskipTests -Dversion.org.wildfly.core=<insert_SNAPSHOT_VERSION_OF_WILDFLY-CORE>
+```
+The `-DskipTests` tag skips all the tests, since we only need to build the projects. You can also go in and manually update the versions in the main pom.xml files of the projects and then use maven to build them. 
+
+=== Build the Docker Image for Quay
+In order to deploy our application on OpenShift, we need to first create a container image of the application using Docker or Podman. In this post we are using Docker, but you can also use Podman using the `podman` keyword, or using an alias.  
+
+Navigate to the `WildFly` project and create a `Dockerfile` inside the root directory of the project. We will be creating an image of the current version of WildFly using the `wildfly-runtime-jdk` image on Quay. Quay is a repository for building and storing container images of your projects. 
+
+Head over to https://quay.io/repository/[Quay.io] and create a repository called `wildfly`. For more information on how to create one, visit the https://docs.quay.io/guides/create-repo.html#:~:text=via%20the%20UI-,To%20create%20a%20repository%20in%20the%20Quay.io%20UI%2C%20click,the%20'Create%20Repository'%20button[Quay documentation].
+
+Now we can create the container image for our WildFly project. Enter the following commands inside your Dockerfile: 
+
+```
```

Comment:
Instead of a ```block```, you could use:

```
[source,bash]
----
FROM quay.io/wildfly/wildfly-runtime-jdk11:latest
COPY --chown=jboss:root dist/target/wildfly-31.0.0.Beta1-SNAPSHOT $JBOSS_HOME
RUN chmod -R ug+rwX $JBOSS_HOME
----
```

Reply:
fixed!